### PR TITLE
Restore support for URL installPackagePath

### LINKED ISSUE #21155: fetch charts from URL not working
This was accidentally removed as part of a refactor. It's required by istioctl upgrade command to check for changed default values.

### PR DESCRIPTION
**Pull request by ostromart**
This restores the ability to specify http[s] targets for installPackagePath which is an Istio release tar format and adds a few fixes. Fixes https://github.com/istio/istio/issues/21155.

### CODE REVIEW

**Review by richardwxn — APPROVED** [review]

File: operator/cmd/mesh/profile-common.go [hunk 1]
```diff
@@ -286,3 +300,15 @@ func unmarshalAndValidateIOPS(iopsYAML string, force bool, l *Logger) (*v1alpha1
 	}
 	return iops, nil
 }
+
+// getInstallPackagePath returns the installPackagePath in the given IstioOperator YAML string.
+func getInstallPackagePath(iopYAML string) (string, error) {
+	iop, err := validate.UnmarshalIOP(iopYAML)
+	if err != nil {
+		return "", nil
```
Comment (elfinhe) [inline operator/cmd/mesh/profile-common.go]: lint found something:
`return "", err`

**Review by elfinhe — APPROVED** [review]
LGTM

**Review by elfinhe — APPROVED** [review]

**Pull request merged**